### DISCUSSION TITLE
Sync playlist button

### DISCUSSION THREAD

**Discussion by shayaantx** [UnicornsOnLSD/finamp#529] [General]
Hello,

Cool app

I added a sync playlist button that only shows up for playlists cause the sync downloads button was nice, but the lack of removing songs from a playlist and having to go to downloads page every time got annoying (rather just go to playlist, press sync, and let it do its magic) - see https://github.com/shayaantx/finamp/commit/f6527e257b450546a6f71e94228951f643c2ffd0

If its something yall think would benefit, I can put up a PR, otherwise feel free to close this discussion

**Comment by Chaphasilor** [comment]
That seems like a nice change! What happens if I downloaded an album with a song and a playlist that contains the same song, then remove the song from the playlist and tap the sync button?  
Ideally the song should stay downloaded as part of the album :)

Otherwise please go ahead and create a PR so we can talk about the implementation details some more! 😁

**Reply by shayaantx** [reply]
I think it "should" not be removed from downloads based on the requiredBy logic I saw in the downloads helper (I'll test it out)

I'll open a PR up and we can clean it up (probably be later this weekend)

fyi

```
    await downloadsHelper.deleteDownloads(jellyfinItemIds: idsToRemove.toList());
    await downloadsHelper.removeChildFromParent(parentId: playlistParentId, childIds: idsToRemove.toList());
```
the removeChildFromParent() is the only "hack" I'd say so far, basically the _downloadedParentsBox is used in downloads page to show each playlists list of songs. So after I removed a song from a playlist in the app, it would still appear in that list (with missing size info).
